pyomo/core/base/constraint.py

```diff
@@ -498,111 +495,86 @@ def set_value(self, expr):
         # user did ( var < 1 > 0 ) (which also results in a non-None
         # chainedInequality value)
         #
-        if EXPR.generate_relational_expression.chainedInequality is not None:
-            raise TypeError(EXPR.chainedInequalityErrorMessage())
+        if EXPR._using_chained_inequality and EXPR._chainedInequality.prev is not None:
+            raise TypeError(EXPR._chainedInequality.error_message())
```

Comment:
Why the first half of the test?  Is this because you only conditionally include the class definition?  I still think that is not a good style.